flags_test.go

```diff
@@ -357,3 +361,117 @@ func (t *FlagsTest) TestValidateFlagsForValidSequentialReadSizeAndHTTP2ClientPro
+			statCacheTTL:             DefaultStatOrTypeCacheTTL,
+			typeCacheTTL:             DefaultStatOrTypeCacheTTL,
+			ttlInSeconds:             config.TtlInSecsUnsetSentinel,
+			expectedMetadataCacheTTL: DefaultStatOrTypeCacheTTL,
+		},
+		{
+			// scenario where user sets only metadata-cache:ttl-secs and sets it to -1
+			statCacheTTL:             DefaultStatOrTypeCacheTTL,
+			typeCacheTTL:             DefaultStatOrTypeCacheTTL,
+			ttlInSeconds:             -1,
+			expectedMetadataCacheTTL: time.Duration(math.MaxInt64),
+		},
+		{
+			// scenario where user sets only metadata-cache:ttl-secs and sets it to 0
+			statCacheTTL:             DefaultStatOrTypeCacheTTL,
+			typeCacheTTL:             DefaultStatOrTypeCacheTTL,
+			ttlInSeconds:             0,
+			expectedMetadataCacheTTL: 0,
+		},
+		{
+			// scenario where user sets only metadata-cache:ttl-secs and sets it to a positive value
+			statCacheTTL:             DefaultStatOrTypeCacheTTL,
+			typeCacheTTL:             DefaultStatOrTypeCacheTTL,
+			ttlInSeconds:             30,
+			expectedMetadataCacheTTL: 30 * time.Second,
+		},
+		{
+			// scenario where user sets only metadata-cache:ttl-secs and sets it to its highest supported value
+			statCacheTTL: DefaultStatOrTypeCacheTTL,
+			typeCacheTTL: DefaultStatOrTypeCacheTTL,
+			ttlInSeconds: config.MaxSupportedTtlInSeconds,
+
+			// math.MaxInt64 is not a multiple of 1e9, so time.Duration(math.MaxInt64) is not
+			// an exact multiple of seconds. Since ttlInSeconds will always set time in
+			// multiple of seconds, we also need to round time.Duration(math.MaxInt64) to
+			// multiple of seconds, for comparison.
+			expectedMetadataCacheTTL: time.Duration((math.MaxInt64 / int64(time.Second)) * int64(time.Second)),
+		},
+		{
+			// scenario where user sets both the old flags and the metadata-cache:ttl-secs. Here ttl-secs overrides both flags. case 1.
+			statCacheTTL:             5 * time.Minute,
+			typeCacheTTL:             time.Hour,
+			ttlInSeconds:             10800,
+			expectedMetadataCacheTTL: 3 * time.Hour,
+		},
+		{
+			// scenario where user sets both the old flags and the metadata-cache:ttl-secs. Here ttl-secs overrides both flags. case 2.
+			statCacheTTL:             5 * time.Minute,
+			typeCacheTTL:             time.Hour,
+			ttlInSeconds:             1800,
+			expectedMetadataCacheTTL: 30 * time.Minute,
+		},
+		{
+			// scenario where user sets both the old flags and the metadata-cache:ttl-secs. Here ttl-secs overrides both flags. case 3.
+			statCacheTTL:             5 * time.Minute,
+			typeCacheTTL:             time.Hour,
+			ttlInSeconds:             0,
+			expectedMetadataCacheTTL: 0,
+		},
+		{
+			// scenario where user sets both the old flags and the metadata-cache:ttl-secs. Here ttl-secs overrides both flags. case 4.
+			statCacheTTL:             5 * time.Minute,
+			typeCacheTTL:             time.Hour,
+			ttlInSeconds:             -1,
```

Comment:
it is okay to not have case 3 and case 4 because these cases are covered above when stat=type-ttl= DefaultStatOrTypeCacheTTL and the function that we are unit testing is not aware of DefaultStatOrTypeCacheTTL.